independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/BootstrapAppModelResolver.java

```diff
@@ -130,24 +143,42 @@ public AppModel resolveModel(AppArtifact appArtifact, List<AppDependency> direct
         return resolveManagedModel(appArtifact, directDeps, null);
     }
 
-    public AppModel resolveManagedModel(AppArtifact appArtifact, List<AppDependency> directDeps, AppArtifact managingProject) throws AppModelResolverException {
+    @Override
+    public AppModel resolveManagedModel(AppArtifact appArtifact, List<AppDependency> directDeps, AppArtifact managingProject)
+            throws AppModelResolverException {
         return doResolveModel(appArtifact, toAetherDeps(directDeps), managingProject);
     }
 
-    private AppModel doResolveModel(AppArtifact appArtifact, List<Dependency> directMvnDeps, AppArtifact managingProject) throws AppModelResolverException {
+    private AppModel doResolveModel(AppArtifact appArtifact, List<Dependency> directMvnDeps, AppArtifact managingProject)
+            throws AppModelResolverException {
+        AppModel.Builder appBuilder = new AppModel.Builder();
         List<Dependency> managedDeps = Collections.emptyList();
         List<RemoteRepository> managedRepos = Collections.emptyList();
-        if(managingProject != null) {
+        if (managingProject != null) {
             final ArtifactDescriptorResult managingDescr = mvn.resolveDescriptor(toAetherArtifact(managingProject));
             managedDeps = managingDescr.getManagedDependencies();
             managedRepos = mvn.newResolutionRepositories(managingDescr.getRepositories());
         }
-
-        DependencyNode resolvedDeps = mvn.resolveManagedDependencies(toAetherArtifact(appArtifact),
-                directMvnDeps, managedDeps, managedRepos, devmode ? new String[] { "test" } : new String[0]).getRoot();
+        List<String> excludedScopes = new ArrayList<>();
+        if (!test) {
+            excludedScopes.add("test");
+        }
+        if (!devmode) {
+            excludedScopes.add("provided");
+        }
 
         final Set<AppArtifactKey> appDeps = new HashSet<>();
         final List<AppDependency> userDeps = new ArrayList<>();
+        DependencyNode resolvedDeps;
+        if (appArtifact != null) {
+            resolvedDeps = mvn.resolveManagedDependencies(toAetherArtifact(appArtifact),
+                    directMvnDeps, managedDeps, managedRepos, excludedScopes.toArray(new String[0])).getRoot();
+        } else {
+            //if there is no main artifact we assume we already have all the deps we need
+            //we just turn them into a DependencyNode
+            resolvedDeps = mvn.toDependencyTree(directMvnDeps, managedRepos).getRoot();
```

Comment:
`directMvnDeps` are supposed to be only the direct deps, their transitive deps aren't in there.